web/src/lottie-player.ts

```diff
@@ -0,0 +1,577 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { html, PropertyValueMap, LitElement, type TemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+// @ts-ignore: WASM Glue code doesn't have type & Only available on build progress
+import Module from '../dist/thorvg-wasm';
+import { THORVG_VERSION } from './version';
+
+let _tvg: any;
+let _module: any;
+(async () => {  
+  _module = await Module();
+  _tvg = new _module.TvgLottieAnimation();
+})();
+
+// Define library version
+export interface LibraryVersion {
+  THORVG_VERSION: string
+}
+
+// Define file type which can be exported
+export enum ExportableType {
+  GIF = 'gif',
+}
+
+// Define valid player states
+export enum PlayerState {
+ Destroyed = "destroyed", // Player is destroyed by `destroy()` method
+ Error = "error", // An error occurred
+ Loading = "loading", // Player is loading
+ Paused = "paused", // Player is paused
+ Playing = "playing", // Player is playing
+ Stopped = "stopped",  // Player is stopped
+ Frozen = "frozen", // Player is paused due to player being invisible
+}
+
+// Define play modes
+export enum PlayMode {
+  Bounce = "bounce",
+  Normal = "normal",
+}
+
+// Define player events
+export enum PlayerEvent {
+  Complete = "complete",
+  Destroyed = "destroyed",
+  Error = "error",
+  Frame = "frame",
+  Freeze = "freeze",
+  Load = "load",
+  Loop = "loop",
+  Pause = "pause",
+  Play = "play",
+  Ready = "ready",
+  Stop = "stop",
+  }
+
+/**
+ * Parse a resource into a bytes
+ */
+const _parseSrc = async (src: string | object): Promise<Uint8Array> => {
+  let data = src;
+  if (typeof data === "object") {
+    data = JSON.stringify(data);
+  } else if (typeof data === "string") {
+    try {
+      data = JSON.parse(data);
+    } catch (err) {
+      const json = await _fromURL(data as string);
+      data = JSON.stringify(json);
+    }
+  }
+
+  const encoder = new TextEncoder();
+  return encoder.encode(data as string);
+}
```

Comment:
I simplified it by moving some validation to the `_fromURL` function, but the fn was renamed to a better name `getLottieFromUrl`, see below.

```suggestion
const _parseSrc = async (src: unknown): Promise<Uint8Array> => {
      const lottie = await getLottieFromUrl(src);
      const lottieStr = JSON.stringify(lottie);
      const encoder = new TextEncoder();

      return encoder.encode(lottieStr);
}
```

Reply:
I'm not sure if it's nice option to use `unknown` in typescript, I try to use less `unknown` and `any`.

Also in the case of src is object, getLottieFromUrl doesn't have to be called, it's json object itself. (to keep less call stack as possible)

Reply:
1. `unknown` is much better than `any` in this case, let me know if you'd want me to share reference docs. You have to use some type if you want to assert an unknown prop that you are validating
2. I didn't bring any new logic into your code base, I simply refactored existing logic and moved some validation from parseSrc to fromUrl for better readability, removing unnecessary variables, and reducing the chance of having unexpected bugs when the code base will be grown